app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1932,5 +1932,169 @@ class PatientRepositoryAndroidTest {
+        recordPatientAtFacility(
+            UUID.fromString("072df9ac-6a00-462f-b87e-1a9ca3b24c1b"),
+            "Patient D",
+            PatientStatus.ACTIVE,
+            facilityB
+        )
+    )
+
+    // when
+    val allPatientsInFacilityA = patientUuidsInAFacility(facilityA)
+    val allPatientsInFacilityB = patientUuidsInAFacility(facilityB)
+    val allPatientsInFacilityC = patientUuidsInAFacility(facilityC)
+
+    // then
+    assertThat(allPatientsInFacilityA)
+        .containsExactly(patientsInFacilityA[1].patient.uuid, patientsInFacilityA[0].patient.uuid)
+        .inOrder()
+
+    assertThat(allPatientsInFacilityB)
+        .containsExactly(patientsInFacilityB[2].patient.uuid)
+        .inOrder()
+
+    assertThat(allPatientsInFacilityC).isEmpty()
+  }
+
+  @Test
+  fun querying_all_patients_in_a_facility_with_bps_in_different_facilities_should_set_the_last_bp_to_the_latest_one() {
+
+    fun recordPatientWithBps(
+        patientUuid: UUID,
+        patientName: String,
+        recordBpsAt: List<Triple<UUID, Facility, Instant>>
+    ): PatientProfile {
+      val patientProfile = testData.patientProfile(patientUuid = patientUuid).let { patientProfile ->
+        patientProfile.copy(patient = patientProfile.patient.copy(status = PatientStatus.ACTIVE, fullName = patientName))
+      }
+
+      val bps = recordBpsAt.map { (uuid, facility, recordedAt) ->
+        testData.bloodPressureMeasurement(uuid = uuid, patientUuid = patientUuid, facilityUuid = facility.uuid, recordedAt = recordedAt)
+      }
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+      bloodPressureRepository.save(bps).blockingAwait()
+
+      return patientProfile
+    }
+
+    fun allPatientIdsAndLastBpRecordedAtFacility(facility: Facility): List<Pair<UUID, Instant>> {
+      return patientRepository.allPatientsInFacility(facility).blockingFirst().map { it.uuid to it.lastBp!!.takenOn }
+    }
+
+    // given
+    val (facilityA, facilityB) = facilityRepository.facilities().blockingFirst()
+    val now = Instant.parse("2019-06-26T00:00:00Z")
+
+    val bpUuid1ForPatientA = UUID.fromString("588b0d8c-6dce-4664-863f-8c79c06c6c3c")
+    val bpUuid2ForPatientA = UUID.fromString("ba875b98-4ca0-4097-bbc7-56041b377cf2")
+    val bpUuid3ForPatientA = UUID.fromString("4b92601e-4a86-40de-b6b7-707f9122a81b")
+
+    val patientWithLatestBpInFacilityA = recordPatientWithBps(
+        UUID.fromString("e2a49529-3b46-4c9e-a7bc-af3090f1ecbb"),
+        "Patient A",
+        listOf(
+            Triple(bpUuid1ForPatientA, facilityA, now.plusSeconds(1)),
+            Triple(bpUuid2ForPatientA, facilityB, now.minusSeconds(1)),
+            Triple(bpUuid3ForPatientA, facilityA, now)
+        )
+    )
+
+    val bpUuid1ForPatientB = UUID.fromString("00e39456-f5e4-4538-956d-7e973ec5da88")
+    val bpUuid2ForPatientB = UUID.fromString("b72e28cd-1b46-4c72-84cd-58530c2829e3")
+    val bpUuid3ForPatientB = UUID.fromString("c18b41d5-260e-4a26-8030-45c621ded98d")
```

Comment:
I dont see any of these generated Ids being used further in the method.
We can leave Ids to be auto-generated.
Same for `Patient A` as well.

Reply:
See above comment about reducing the amount of random data in our tests. I think @ragunathjawahar will be able to explain this decision in a better way than I can.